graylog2-server/src/main/java/org/graylog2/plugin/Message.java

```diff
@@ -432,40 +421,45 @@ public DateTime getTimestamp() {
         obj.put(FIELD_STREAMS, getStreamIds());
         obj.put(FIELD_GL2_ACCOUNTED_MESSAGE_SIZE, getSize());
 
+        final Object timestampValue = getField(FIELD_TIMESTAMP);
+        DateTime dateTime = convertToDateTime(timestampValue);
+        if (dateTime != null) {
+            obj.put(FIELD_TIMESTAMP, buildElasticSearchTimeFormat(dateTime.withZone(UTC)));
+        }
```

Comment:
I find it a bit confusing that we are "converting" the timestamp again. If I'm not missing anything, it's either already in the right format at this point, or it's `null`. So I'd prefer to explicitly handle the `null` case here. I think it would make it more obvious what's going on.

Reply:
I think I agree :thinking:  But how would you write that without duplicating too much code?

We could also drop that null check, because `convertToDateTime` never returns `null`
```suggestion
        final Object timestampValue = getField(FIELD_TIMESTAMP);
        DateTime dateTime = convertToDateTime(timestampValue);
        obj.put(FIELD_TIMESTAMP, buildElasticSearchTimeFormat(dateTime.withZone(UTC)));
```

Reply:
🤔  maybe like so?
```
Index: graylog2-server/src/main/java/org/graylog2/plugin/Message.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/graylog2-server/src/main/java/org/graylog2/plugin/Message.java b/graylog2-server/src/main/java/org/graylog2/plugin/Message.java
--- a/graylog2-server/src/main/java/org/graylog2/plugin/Message.java	(revision 473e376b96308f200e97d7fdcc5fca78e7555098)
+++ b/graylog2-server/src/main/java/org/graylog2/plugin/Message.java	(date 1631031513849)
@@ -422,10 +422,8 @@
         obj.put(FIELD_GL2_ACCOUNTED_MESSAGE_SIZE, getSize());
 
         final Object timestampValue = getField(FIELD_TIMESTAMP);
-        DateTime dateTime = convertToDateTime(timestampValue);
-        if (dateTime != null) {
-            obj.put(FIELD_TIMESTAMP, buildElasticSearchTimeFormat(dateTime.withZone(UTC)));
-        }
+        DateTime dateTime = timestampValue == null ? fallbackForNullTimestamp() : convertToDateTime(timestampValue);
+        obj.put(FIELD_TIMESTAMP, buildElasticSearchTimeFormat(dateTime.withZone(UTC)));
 
         if (processingErrors != null && !processingErrors.isEmpty()) {
             if (processingErrors.stream().anyMatch(processingError -> processingError.getCause().equals(ProcessingFailureCause.InvalidTimestampException))) {
@@ -441,15 +439,6 @@
     }
 
     private DateTime convertToDateTime(Object value) {
-        if (value == null) {
-            final String error = "<null> value for field timestamp in message <" + getId() + ">, forcing to current time";
-            LOG.trace(error);
-            addProcessingError(new ProcessingError(ProcessingFailureCause.InvalidTimestampException,
-                    "Replaced invalid timestamp value in message <" + getId() + "> with current time",
-                    "<null> value provided"));
-            return Tools.nowUTC();
-        }
-
         try {
             return DateTimeConverter.convertToDateTime(value);
         } catch (IllegalArgumentException e) {
@@ -461,6 +450,15 @@
             return Tools.nowUTC();
         }
     }
+
+    private DateTime fallbackForNullTimestamp() {
+        final String error = "<null> value for field timestamp in message <" + getId() + ">, forcing to current time";
+        LOG.trace(error);
+        addProcessingError(new ProcessingError(ProcessingFailureCause.InvalidTimestampException,
+                "Replaced invalid timestamp value in message <" + getId() + "> with current time",
+                "<null> value provided"));
+        return Tools.nowUTC();
+    }
 
     // estimate the byte/char length for a field and its value
     static long sizeForField(@Nonnull String key, @Nonnull Object value) {
@@ -536,7 +534,7 @@
 
         final boolean isTimestamp = FIELD_TIMESTAMP.equals(trimmedKey);
         if (isTimestamp) {
-            final DateTime timeStamp = convertToDateTime(value);
+            final DateTime timeStamp = value == null ? fallbackForNullTimestamp() : convertToDateTime(value);
             final Object previousValue = fields.put(FIELD_TIMESTAMP, timeStamp);
             updateSize(trimmedKey, timeStamp, previousValue);
         } else if (value instanceof String) {

```

Reply:
Thanks :-) Applied :+1: